pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1001,20 +1009,30 @@ type SemaphoreHolding struct {
 	Holders []string `json:"holders,omitempty" protobuf:"bytes,2,opt,name=holders"`
 }
 
-type WaitingStatus struct {
-	// Holders stores the list of current holder names
-	Holders HolderNames `json:"holders,omitempty" protobuf:"bytes,1,opt,name=holders"`
+// MutexHolding holds the resource that holds the mutex lock.
```

Comment:
maybe just `the resources that holds the mutex`

can this be any workflow in the namespace? can it be something other that a workflow? can it be this workflow or always another workflow?